Valet/VALSecureEnclaveValet.m

```diff
@@ -140,47 +300,115 @@ - (nullable NSError *)migrateObjectsMatchingQuery:(nonnull NSDictionary *)secIte
-    NSMutableDictionary *mutableBaseQuery = [super mutableBaseQueryWithIdentifier:identifier initializer:initializer accessibility:accessibility];
-    
-    // Add the access control, which opts us in to Secure Element storage.
-    mutableBaseQuery[(__bridge id)kSecAttrAccessControl] = (__bridge_transfer id)SecAccessControlCreateWithFlags(NULL, kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly, kSecAccessControlUserPresence, NULL);
+    // Remove the key before trying to set it. This will prevent us from calling SecItemUpdate on an item stored on the Secure Enclave, which would cause iOS to prompt the user for authentication.
+    [self removeObjectForKey:key];
     
-    // kSecAttrAccessControl and kSecAttrAccessible are mutually exclusive, so remove kSecAttrAccessible from our query.
-    [mutableBaseQuery removeObjectForKey:(__bridge id)kSecAttrAccessible];
+    return [super setObject:value forKey:key options:options];
+}
+
+@end
+
+
+#pragma mark - Deprecated Category
+
+
+@implementation VALSecureEnclaveValet (Deprecated)
+
+#pragma mark - Deprecated Initializers
+
+- (nullable instancetype)initWithIdentifier:(nonnull NSString *)identifier;
+{
+    return [self initWithIdentifier:identifier accessibility:VALAccessibilityWhenPasscodeSetThisDeviceOnly];
+}
+
+- (nullable instancetype)initWithIdentifier:(nonnull NSString *)identifier accessibility:(VALAccessibility)accessibility;
+{
+    VALCheckCondition(accessibility == VALAccessibilityWhenPasscodeSetThisDeviceOnly, nil, @"Accessibility on SecureEnclaveValet must be VALAccessibilityWhenPasscodeSetThisDeviceOnly");
     
-    return mutableBaseQuery;
-#else
-    return [NSMutableDictionary new];
-#endif
+    return [self initWithIdentifier:identifier accessControl:VALAccessControlUserPresence];
 }
 
-- (BOOL)setObject:(nonnull NSData *)value forKey:(nonnull NSString *)key options:(NSDictionary *)options;
+- (nullable instancetype)initWithSharedAccessGroupIdentifier:(nonnull NSString *)sharedAccessGroupIdentifier;
 {
-    // Remove the key before trying to set it. This will prevent us from calling SecItemUpdate on an item stored on the Secure Enclave, which would cause iOS to prompt the user for authentication.
-    [self removeObjectForKey:key];
+    return [self initWithSharedAccessGroupIdentifier:sharedAccessGroupIdentifier accessibility:VALAccessibilityWhenPasscodeSetThisDeviceOnly];
+}
+
+- (nullable instancetype)initWithSharedAccessGroupIdentifier:(nonnull NSString *)sharedAccessGroupIdentifier accessibility:(VALAccessibility)accessibility;
+{
+    VALCheckCondition(accessibility == VALAccessibilityWhenPasscodeSetThisDeviceOnly, nil, @"Accessibility on SecureEnclaveValet must be VALAccessibilityWhenPasscodeSetThisDeviceOnly");
     
-    return [super setObject:value forKey:key options:options];
+    return [self initWithSharedAccessGroupIdentifier:sharedAccessGroupIdentifier accessControl:VALAccessControlUserPresence];
+}
+
+@end
+
+#else // VAL_SECURE_ENCLAVE_SDK_AVAILABLE
```

Comment:
Having `// VAL_SECURE_ENCLAVE_SDK_AVAILABLE` here confuses reading of the below.